### PR TITLE
Fix copy gpu of views

### PR DESCRIPTION
**Pull request by erikzenker**
Fixed cuda pitched ptr calculation for the 3D copy case.

### CODE REVIEW

**Comment by BenjaminW3** [comment]
Thanks! Could you please rename `m_*PitchBytes` to `m_*PitchBytesX`?

**Comment by erikzenker** [comment]
sure

**Comment by erikzenker** [comment]
Fixed and rebased. The CI log is getting to long :cyclone:

File: include/alpaka/mem/buf/cuda/Copy.hpp [hunk 1]
```diff
@@ -564,20 +571,21 @@ namespace alpaka
...
                         //cudaMemCpy3DParms.srcPos;       // Optional. Offset in bytes.
+
                         cudaMemCpy3DParms.srcPtr =
                             make_cudaPitchedPtr(
                                 const_cast<void *>(srcNativePtr),
-                                srcPitchBytes,
+                                srcpitchBytesX,
                                 srcWidth,
-                                srcHeight);
+                                srcPitchBytesY/srcpitchBytesX);
```
Comment (BenjaminW3) [inline include/alpaka/mem/buf/cuda/Copy.hpp]: A lot of trailing spaces...

**Comment by BenjaminW3** [comment]
There is still an error:
`/home/travis/build/ComputationalRadiationPhysics/alpaka/include/alpaka/mem/buf/cuda/Copy.hpp(626): error: class "alpaka::mem::view::cuda::detail::TaskCopy<std::integral_constant<std::size_t, 2UL>, alpaka::mem::buf::BufCpu<uint32_t, std::integral_constant<std::size_t, 2UL>, uint32_t>, alpaka::mem::buf::BufCudaRt<uint32_t, alpaka::dim::DimInt<2UL>, uint32_t>, alpaka::Vec<alpaka::dim::DimInt<2UL>, uint32_t>>" has no member "m_dstPitchBytesY"`
detected during:
`instantiation of "auto alpaka::mem::view::cuda::detail::buildCudaMemcpy3DPeerParms(const alpaka::mem::view::cuda::detail::TaskCopy<alpaka::dim::DimInt<2UL>, TViewDst, TViewSrc, TExtent> &)->cudaMemcpy3DPeerParms [with TViewDst=alpaka::mem::buf::BufCpu<uint32_t, std::integral_constant<std::size_t, 2UL>, uint32_t>, TViewSrc=alpaka::mem::buf::BufCudaRt<uint32_t, alpaka::dim::DimInt<2UL>, uint32_t>, TExtent=alpaka::Vec<alpaka::dim::DimInt<2UL>, uint32_t>]"`
What this tells you is that the 2D `TaskCopy` does not have a `m_dstPitchBytesY` member.
Because there is no CUDA peer memcpy for 2D but only for 3D, the 2D implementation uses the 3D CUDA peer memcpy.
To make this work you either have to add the y pitches to the 2D `TaskCopy` or you have to add a `buildCudaMemcpy3DPeerParmsFrom2D` method that does not try to access those members.
The first option would be much easier because you can directly copy the members and their initialization from the 3D `TaskCopy`.

**Comment by erikzenker** [comment]
I added the `m_dstPitchBytesY` member to the 2D `TaskCopy` class.

**Pull request merged**